app/models/account.rb

```diff
@@ -52,18 +52,12 @@ class Account < ApplicationRecord
   has_one :user, inverse_of: :account
 
   validates :username, presence: true
-
-  # Remote user validations
-  with_options unless: :local? do
-    validates :username, uniqueness: { scope: :domain, case_sensitive: true }, if: :username_changed?
-  end
+  validates :username, uniqueness: { scope: :domain, case_sensitive: true }, if: :username_changed?
 
   # Local user validations
-  with_options if: :local? do
-    validates :username, format: { with: /\A[a-z0-9_]+\z/i }, uniqueness: { scope: :domain, case_sensitive: false }, length: { maximum: 30 }, unreserved: true, if: :username_changed?
-    validates :display_name, length: { maximum: 30 }, if: :display_name_changed?
-    validates :note, length: { maximum: 160 }, if: :note_changed?
-  end
+  validates :username, format: { with: /\A[a-z0-9_]+\z/i }, length: { maximum: 30 }, unreserved: true, if: -> { local? && username_changed? }
```

Comment:
I think the `with_options` usage here was more clear and this change is awkward.